torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -688,7 +689,9 @@ Tensor* TensorExprKernel::computeThreeOperand(
             tensorOrConstant(n->inputs()[2], indices),
         };
 
-        promoteInputs(inputs);
+        if (promote_inputs) {
```

Comment:
Should we still be demoting output with this flag?

Reply:
Yes, if I understand correctly the output type should still be unchanged.